packages/roc-package-web-app-react/app/server/render.js

```diff
@@ -99,13 +96,11 @@ export function reactRender(url, createRoutes, store, renderPage, staticRender =
                     };
                 }
 
-                getPrefetchedData(components, locals)
+                trigger('fetch', components, locals)
                     .then(() => {
-                        let component = <RoutingContext {...renderProps} />;
+                        let component = <RouterContext {...renderProps} />;
```

Comment:
I'm not sure what you mean here, what do you want to achieve? _I have created a suggested API below that might offer what is requested._

>additional fetching methods, such as optional data fetches that are allowed to fail and also triggering the deferred data fetches on the server.

Like creating additional things that needs to be completed before rendering the page? So an API for this could look like this?

```js
const extra = userDefinedThings.map((userDefinedThing) => userDefinedThing({
  dispatch,
  getState,
  location,
  params,
  components
})

Promise.all([trigger, ...extra]).then(() => { //...
```

> triggering the deferred data fetches on the server
Why would we want to perform the `defer` data fetching on the server?


I have also looked into some other ways to modify what the server does based on our discussion from earlier. This makes it possible to 
- Define extra things on `locals` that is used for `trigger()`
_Not sure that this is a good idea since we might have a miss match from what the client have and more checks would need to be performed the function in both contexts_
- Use things like `react-side-effects` and other things that should modify the template.

````js
// "Pseudo" code
match(..., () => {
  // Do checks about if the path was "valid"
  const { locals } = runBeforeRender({ store }) // Can perform dispatch and give back locals
  
  const page = ReactDOM.renderToString()
  
  const { templateProps } = runAfterRender({ getState }) // Would be possible to modify store here but is that a good idea?
  
  return renderPage(page, templateProps)
}) 
```
In this example runAfterRender will be able to do for instance `.rewind()` on some singleton thing.